### PR TITLE
P1 components 

### PR DESCRIPTION
**Pull request by foolhardy21**
This PR has demos of:
- Toast component.
- Grids structure.
- Text Utilities.
- Modal component.
- Navigation component.
- List component.
- Rating component.

Feel free to review. Thanks.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -4,16 +4,54 @@ const logoBtn = document.querySelector('.logo')
...
-const driftLogo = document.querySelector('#drift-logo') 
+
 const topScrollBtn =  document.querySelector('.float-btn') 
+
 const formSubmit = document.querySelector('#submit-btn')
 const formInput = document.querySelector('#input')
+
 const hideMenuBtn = document.querySelector('#hide-drawer')
 const showMenuBtn = document.querySelector('#show-drawer')
 const Menu = document.querySelector('#drawer')
```
Comment (Vishnu-Aithal) [inline script.js:12]: small Typo Menu Variable

File: script.js [hunk 2]
```diff
@@ -4,16 +4,54 @@ const logoBtn = document.querySelector('.logo')
...
+            } else {
+                star.style.color = 'var(--c-gray)' 
+            }
+        })
+    })
+})
+
 showMenuBtn.addEventListener('click', () => {
     Menu.style.display = 'block'
     showMenuBtn.style.display = 'none'
```
Comment (Vishnu-Aithal) [inline script.js:57]: Suggestion:  Show Menu By Defualt For Bigger Screens
Example Code::

window.onload = () => {
  if (matchMedia("only screen and (min-width:600px)").matches) {
       menu.style.display = "block";
       showMenuBtn.style.display = "none"
  }
}

Comment (Vishnu-Aithal) [reply script.js:57]: or better to implement in css

**Review by Vishnu-Aithal — APPROVED** [review]
suggestion : - Hide Menu By default on smaller screens
                    - Uniform margins when menu Hidden
                    - Transparent bg to showMenu button, opaque on hover for better visibility

**Review by anik31 — APPROVED** [review]
Looks good to me. Suggestion - add some space between documentation & component list

**Comment by foolhardy21** [comment]
Thanks @anik31 and @Vishnu-Aithal. I have added the changes mentioned. For now I'm keeping the menu shown by default for all screens as the user can hide it anytime using the button.

**Pull request merged**